### PR TITLE
nctl: add test to check era summary RPCs after upgrade

### PR DESCRIPTION
**Pull request by alsrdn**
Add test to check the backward compatibility of
`chain_get_era_info_by_switch_block` RPC endpoint.

### CODE REVIEW

File: utils/nctl/sh/scenarios-upgrades/upgrade_scenario_13.sh [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+    local LATEST_SWITCH_BLOCK_HASH=$(echo $LATEST_SWITCH_BLOCK | jq '.hash')
+    _compare_era_summary $LATEST_SWITCH_BLOCK_HASH
+
+    # check that chain_get_era_info_by_switch_block still works for blocks before the upgrade
+    local SWITCH_BLOCK_BEFORE_UPGRADE=$(get_switch_block "1" "150" "" "2")
+    local SWITCH_BLOCK_BEFORE_UPGRADE_HASH=$(echo $SWITCH_BLOCK_BEFORE_UPGRADE | jq '.hash')
+    _compare_era_summary $SWITCH_BLOCK_BEFORE_UPGRADE_HASH
+}
+
+# Step 09: Terminate.
```
Comment (Fraser999) [inline utils/nctl/sh/scenarios-upgrades/upgrade_scenario_13.sh]:
```suggestion
# Step 08: Terminate.
```

File: utils/nctl/sh/scenarios-upgrades/upgrade_scenario_13.sh [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+}
+
+# Step 06: Await 2 eras.
+function _step_06()
+{
+    log_step_upgrades 6 "awaiting 2 eras"
+    await_n_eras '2' 'true' '5.0' '2'
+}
+
+function _compare_era_summary()
```
Comment (Fraser999) [inline utils/nctl/sh/scenarios-upgrades/upgrade_scenario_13.sh:154]: I think we should also check in here that the returned `.result.era_summary` is not `null` - either in this function or in a separate one if you prefer.

File: utils/nctl/sh/scenarios-upgrades/upgrade_scenario_13.sh [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+# Check rpc backwards compatibility for the chain_get_era_info_by_switch_block RPC
+function _step_07()
+{
+    log_step_upgrades 7 "comparing era summary obtained through chain_get_era_info_by_switch_block and chain_get_era_summary"
+
+    local LATEST_SWITCH_BLOCK=$(get_switch_block "1" "32" "" "")
+    local LATEST_SWITCH_BLOCK_HASH=$(echo $LATEST_SWITCH_BLOCK | jq '.hash')
+    _compare_era_summary $LATEST_SWITCH_BLOCK_HASH
+
+    # check that chain_get_era_info_by_switch_block still works for blocks before the upgrade
```
Comment (Fraser999) [inline utils/nctl/sh/scenarios-upgrades/upgrade_scenario_13.sh]:
```suggestion
    # check both RPCs still work for switch blocks before the upgrade
```

File: utils/nctl/sh/scenarios-upgrades/upgrade_scenario_13.sh (hunk 1 above)
Comment (alsrdn) [reply utils/nctl/sh/scenarios-upgrades/upgrade_scenario_13.sh]: Fixed in https://github.com/Fraser999/casper-node/pull/111/commits/271fc69ce19fe8b88f6273106151024524e884e8

File: utils/nctl/sh/scenarios-upgrades/upgrade_scenario_13.sh (hunk 2 above)
Comment (alsrdn) [reply utils/nctl/sh/scenarios-upgrades/upgrade_scenario_13.sh:154]: Added check in https://github.com/Fraser999/casper-node/pull/111/commits/271fc69ce19fe8b88f6273106151024524e884e8

File: utils/nctl/sh/scenarios-upgrades/upgrade_scenario_13.sh (hunk 3 above)
Comment (alsrdn) [reply utils/nctl/sh/scenarios-upgrades/upgrade_scenario_13.sh]: Fixed in https://github.com/Fraser999/casper-node/pull/111/commits/271fc69ce19fe8b88f6273106151024524e884e8

File: utils/nctl/sh/scenarios-upgrades/upgrade_scenario_13.sh [hunk 4]
```diff
@@ -0,0 +1,247 @@
...
+            }" | jq '.result.era_summary'
+    )
+
+    if [ "$ERA_SUMMARY_RPC_RESPONSE" = "null" ]; then
+        log "ERROR :: chain_get_era_summary failed with block hash $BLOCK_HASH"
+        exit 1
+    fi
+
+    if [ "$ERA_INFO_RPC_RESPONSE" = "null" ]; then
+        log "ERROR :: chain_get_era_summary failed with block hash $BLOCK_HASH"
```
Comment (Fraser999) [inline utils/nctl/sh/scenarios-upgrades/upgrade_scenario_13.sh]:
```suggestion
        log "ERROR :: chain_get_era_info_by_switch_block failed with block hash $BLOCK_HASH"
```

**Review by Fraser999 — APPROVED** [review]

**Pull request merged**